src/Microsoft.OpenApi/Services/CopyReferences.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Services
+{
+    internal class CopyReferences : OpenApiVisitorBase
+    {
+        private readonly OpenApiDocument target;
+        public OpenApiComponents Components = new OpenApiComponents();
+
+        public CopyReferences(OpenApiDocument target)
+        {
+            this.target = target;
```

Comment:
You won't need the `this` prefix if you rename the private field as suggested above.